File: backend/database/models/IrrigatePredictions.js
const { Model, DataTypes } = require('sequelize');

const sequelize = require('../sequelize');
const FarmableLand = require('./FarmableLand');

class IrrigatePredictions extends Model {}

IrrigatePredictions.init(
  {
    date: {
      type: DataTypes.DATE,
      primaryKey: true,
    },
    FarmableLandId: {
      type: DataTypes.BIGINT,
      primaryKey: true,
    },
    amountWater: DataTypes.DOUBLE,
    lengthMinutes: DataTypes.DOUBLE,
  },
  {
    sequelize,
    modelName: 'IrrigatePredictions',
    freezeTableName: true,
    tableName: 'IrrigatePredictions',
    timestamps: false,
  }
);

IrrigatePredictions.belongsTo(FarmableLand);

module.exports = IrrigatePredictions;

File: predictions/models/irrigate_predictions.py
import datetime

from sqlalchemy import Date
from sqlalchemy.orm import Mapped, mapped_column, DeclarativeBase
from sqlalchemy.sql import func


class Base(DeclarativeBase):
    pass


class IrrigatePredictions(Base):
    __tablename__ = "IrrigatePredictions"

    date: Mapped[datetime.date] = mapped_column(
        Date(),
        server_default=func.now(),
        primary_key=True
    )
    lengthMinutes: Mapped[float]
    amountWater: Mapped[float]
    FarmableLandId: Mapped[int]
